mmv1/templates/terraform/examples/dataplex_entry_full.tf.tmpl

```diff
@@ -0,0 +1,133 @@
+  location     = "us-central1"
+  project      = "{{index $.TestEnvVars "project_number"}}"
+
+  metadata_template = <<EOF
+{
+  "name": "tf-test-template",
+  "type": "record",
+  "recordFields": [
+    {
+      "name": "story",
+      "type": "enum",
+      "annotations": {
+        "displayName": "Story",
+        "description": "Specifies the story of an entry."
+      },
+      "index": 1,
+      "constraints": {
+        "required": true
+      },
+      "enumValues": [
+        {
+          "name": "SEQUENCE",
+          "index": 1
+        }
+      ]
+    }
+  ]
+}
+EOF
+}
+
+resource "google_dataplex_entry_group" "{{index $.Vars "entry_group_name"}}" {
+  entry_group_id = "{{index $.Vars "entry_group_name"}}"
+  project = "{{index $.TestEnvVars "project_number"}}"
+  location = "us-central1"
+}
+
+resource "google_dataplex_entry_type" "{{index $.Vars "entry_type_name"}}" {
+  entry_type_id = "{{index $.Vars "entry_type_name"}}"
+  project = "{{index $.TestEnvVars "project_number"}}"
+  location = "us-central1"
+
+  required_aspects {
+    type = google_dataplex_aspect_type.{{index $.Vars "aspect_type_name"}}-one.name
+  }
+}
+
+resource "google_dataplex_entry" "{{$.PrimaryResourceId}}" {
+  entry_group_id = google_dataplex_entry_group.{{index $.Vars "entry_group_name"}}.entry_group_id
+  project = "{{index $.TestEnvVars "project_number"}}"
+  location = "us-central1"
+  entry_id = "{{index $.Vars "entry_id"}}"
+  entry_type = google_dataplex_entry_type.{{index $.Vars "entry_type_name"}}.name
+  fully_qualified_name = "bigquery:{{index $.TestEnvVars "project_number"}}.test-dataset"
+  parent_entry = "projects/{{index $.TestEnvVars "project_number"}}/locations/us-central1/entryGroups/{{index $.Vars "entry_group_name"}}/entries/some-other-entry"
+  entry_source {
+    resource = "bigquery:{{index $.TestEnvVars "project_number"}}.test-dataset"
+    system = "System III"
+    platform = "BigQuery"
+    display_name = "Human readable name"
+    description = "Description from source system"
+    labels = {
+      "some-label": "some-value"
+    }
+
+    ancestors {
+      name = "ancestor-one"
+      type = "type-one"
+    }
+
+    ancestors {
+      name = "ancestor-two"
+      type = "type-two"
+    }
+
+    create_time = "2023-08-03T19:19:00.094Z"
+    update_time = "2023-08-03T20:19:00.094Z"
+  }
+
+  aspects {
+    aspect_key = "{{index $.TestEnvVars "project_number"}}.us-central1.{{index $.Vars "aspect_type_name"}}-one"
+    aspect_value {
+      data = <<EOF
+          {"type": "VIEW"    }
+        EOF
+    }
+  }
+
+  aspects {
+    aspect_key = "{{index $.TestEnvVars "project_number"}}.us-central1.{{index $.Vars "aspect_type_name"}}-two"
+    aspect_value {
+      data = <<EOF
+          {"story": "SEQUENCE"    }
+        EOF
+    }
+  }
+ depends_on = [google_dataplex_aspect_type.{{index $.Vars "aspect_type_name"}}-two, google_dataplex_aspect_type.{{index $.Vars "aspect_type_name"}}-one]
```

Comment:
If you change the `aspect_key` fields to reference the `google_dataplex_aspect_type` resources as in my example above you'll no longer need this `depends_on` as Terraform will understand the order in which the resources need to be created.